### DISCUSSION TITLE
[Node Operator Question] Can we configure max gas limit or block gas in L2 (Testnet)

### DISCUSSION THREAD

**Discussion by priyanshuthapliyal55** [ethereum-optimism/developers#367] [Node Operators] [answered]
### Are you running the most up to date node software?

- [X] Yes

### Did you check the documentation?

- [X] Yes

### Did you check for duplicate questions?

- [X] Yes

### Issue Description

I am deploying L2 rollups testnet on L1 sepolia. Now i want to know if i can configure max gas limit or block gas in L2 rollups. If yes then how or what parameter i can use for that.

### Protocol Description

op-geth - Geth/v1.101311.1-rc.1-84b52dfd-20240423/linux-amd64/go1.21.6
op-node - version v0.0.0-a06cae81-1705510057
L1 chain - ETH Sepolia

**Comment by sbvegan** [comment]
Yes you can configure the max gas limit, see `Gas Limit` in the `Consensus Parameters` table on the [OP Stack Configurability specs](https://specs.optimism.io/protocol/configurability.html). It looks like this parameter is managed by the SystemConfig, this proxy contract is deployed on the L1 and is picked up by L2 as part of the derivation of the L2 chain. This is generally set at genesis, I believe you can update it, but I'm pretty sure this requires a hard fork/network upgrade.

**Reply by priyanshuthapliyal55** [reply]
So I just need to update the confige json file which we use when deploying contracts?

**Reply by sbvegan** [reply, marked as answer]
Yes, I believe the parameter is `l2GenesisBlockGasLimit` and the value is a hex value. Example for `30_000_000` gas

```
"l2GenesisBlockGasLimit": "0x1c9c380",
```

**Reply by priyanshuthapliyal55** [reply]
So by default it is 30_000_000 and by any means is there any max limit that we can change this to ? Or we can change this to any value?

**Reply by sbvegan** [reply]
You can raise you gas limit to whatever your hardware can support. If the blocks are full and you have high limit, it can take a long time for the sequencer to build the block. It might exceed the normal ~2s block time
